src/controllers/confirmation.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { Session } from "ch-node-session-handler";
+import { UserProfileKeys } from "ch-node-session-handler/lib/session/keys/UserProfileKeys";
+import { NextFunction, Request, Response } from "express";
+import { SESSION_OBJECTION_ID } from "../constants";
+import { Templates } from "../model/template.paths";
+import { retrieveFromObjectionSession, retrieveUserProfileFromSession } from "../services/objection.session.service";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+    const session: Session | undefined  = req.session;
+    if (session) {
+        const userProfile = retrieveUserProfileFromSession(session);
+        const email: string = userProfile[UserProfileKeys.Email] as string;
```

Comment:
rather than getting the user profile to get the email from it, why not just have a getEmail function on session service

Reply:
I did consider this but it just seemed better to get the whole user profile back as it has many fields and makes it more reusable.

Reply:
the downside is that we have session knowledge leaking out into the controllers, they will have to have knowledge of the construction of the userProfile. Anyway that's my 2p.

Reply:
Alright I've updated this.